src/neo4clj/query_builder.clj

```diff
@@ -8,19 +8,31 @@
   [{:keys [ref-id] :as node} return?]
   (str "CREATE " (cypher/node node) (when return? (str " RETURN " ref-id))))
 
-(defn lookup-query
-  "Takes a lookup representation and generates a bolt query
-
-  A lookup representation needs the :reference.id to be set and
-  either the :id or :labels and :properties keys"
-  [{:keys [ref-id] :as lookup} return?]
-  (let [cypher-lookup (cypher/lookup lookup)]
+(defn lookup
+  [entity-fn {:keys [ref-id] :as entity} return?]
+  (let [[base-cypher where-cypher] (entity-fn entity)]
     (str "MATCH "
-         (str (first cypher-lookup)
-              (when (second cypher-lookup)
-                (str " WHERE " (second cypher-lookup))))
+         (str base-cypher
+              (when where-cypher
+                (str " WHERE " where-cypher)))
          (when return? (str " RETURN " ref-id)))))
 
+(defn lookup-node
+  "Takes a node lookup representation and generates a bolt query
+
+  A node lookup representation needs the :ref-id to be set and
+  at least one of the keys :id, :labels or :props"
+  [node return?]
+  (lookup cypher/lookup-node node return?))
+
+(defn lookup-relationship
```

Comment:
This is just my personal preference but I think any function being exposed to the "outside" should be named as short as possible (without loosing its meaning of course). How would `lookup-rel` work?

Reply:
Fixed